This patch is based on the upstream commit described below, adapted for use
in the Debian package by Peter Michael Green.

commit 61b8e62f4205b1b2609054bb0d07771b56a5ef9c
Author: Paul Holzinger <pholzing@redhat.com>
Date:   Mon Sep 11 18:13:13 2023 +0200

    bump nix to 0.27.1
    
    The nix crate removes all features by default so we need to explicitly
    list what we use.
    Second, the io API's changed and all use the AsFd trait now. This is
    great for IO safety but will require larger changes to migrate from
    RawFd. Thus I went the easy way and use unsafe (not really unsafe here)
    and construct and a new BorrowedFd object (which impl AsFd) for the
    setns call.
    
    Signed-off-by: Paul Holzinger <pholzing@redhat.com>

Index: netavark/src/network/core_utils.rs
===================================================================
--- netavark.orig/src/network/core_utils.rs
+++ netavark/src/network/core_utils.rs
@@ -280,7 +280,10 @@ impl CoreUtils {
 }
 
 pub fn join_netns(fd: RawFd) -> NetavarkResult<()> {
-    match sched::setns(fd, sched::CloneFlags::CLONE_NEWNET) {
+    match sched::setns(
+        unsafe { BorrowedFd::borrow_raw(fd) },
+        sched::CloneFlags::CLONE_NEWNET,
+    ) {
         Ok(_) => Ok(()),
         Err(e) => Err(NetavarkError::wrap(
             "setns",
